cdap-ui/app/features/adapters/services/etl-create-factory.js

```diff
@@ -32,6 +32,17 @@ angular.module(PKG.name + '.feature.adapters')
       return icon;
     }
 
+    AdapterApiFactory.prototype.fetchTemplates = function() {
+      this.dataSrc.request({
+        _cdapPath: '/templates'
+      })
+        .then(function(res) {
+          this.scope.adapterTypes = res || [];
+          this.scope.metadata.type = this.scope.adapterTypes[0].name;
```

Comment:
The previous line says that `this.scope.adapterTypes` could be `[]`.
In that case, wouldn't `[0]` would fail, right?